### PR TITLE
feat(dotnet,java): kernel process inherits host's STDERR

### PR DESCRIPTION
**Pull request by RomainMuller**
Removed the code that surrounded handling of the `STDERR` stream from
the Kernel process: there is no need to PIPE it, and simply inheriting
it (i.e: subprocess `STDERR` is the same stream as the parent's) results
in a more intuitive behavior, as `STDERR` output from child processes
surfaces "immediately" even if the parent process somehow cannot
forward it anymore (e.g: if it crashed or deadlocked). This also means
the `STDERR` forwarding thread is no longer needed.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs [hunk 1]
```diff
@@ -38,8 +38,7 @@ public NodeProcess(IJsiiRuntimeProvider jsiiRuntimeProvider, ILoggerFactory logg
                     StandardInputEncoding = utf8,
                     RedirectStandardOutput = true,
                     StandardOutputEncoding = utf8,
-                    RedirectStandardError = true,
-                    StandardErrorEncoding = utf8
+                    UseShellExecute = false,
```
Comment (eladb) [inline packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs:41]: I think it's a good change but usually has implications on being able to locate executables. Have you tested this on windows?

File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs [hunk 2]
```diff
@@ -58,18 +57,23 @@ public NodeProcess(IJsiiRuntimeProvider jsiiRuntimeProvider, ILoggerFactory logg
...
+
         public TextWriter StandardInput => _process.StandardInput;
 
         public TextReader StandardOutput => _process.StandardOutput;
 
-        public TextReader StandardError => _process.StandardError;
-
         void IDisposable.Dispose()
         {
-            StandardInput.Dispose();
```
Comment (eladb) [inline packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs:69]: Why are these not needed anymore ?

File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/Runtime.cs [hunk 3]
```diff
@@ -10,19 +9,14 @@ internal sealed class Runtime : IRuntime
         public Runtime(INodeProcess nodeProcess)
         {
             _nodeProcess = nodeProcess ?? throw new ArgumentNullException(nameof(nodeProcess));
-            if (Environment.GetEnvironmentVariable("JSII_DEBUG") != null)
```
Comment (eladb) [inline packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/Runtime.cs:13]: What functionality do we lose here?

**Review by eladb — COMMENTED** [review]
Just to make sure I understand: the child process respects JSII_DEBUG and will only output traces to stderr in that case. So it's safe to just INHERIT stderr, correct?

File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs (hunk 1 above)
Comment (RomainMuller) [reply packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs:41]: * According to [.NET's own documentation](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.redirectstandardoutput?view=net-5.0#remarks), `UseShellExecute = false` is **required** if you set any `Redirect*` to `true`.
* The *default* value of this property is `false` on `.NET Core` (but `true` on `.NET Framework`). This is only making it explicit

File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs (hunk 2 above)
Comment (RomainMuller) [reply packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/NodeProcess.cs:69]: `StandardInput`, `StandardOutput` and `StandardError` are actually directly obtained from properties of the `Process` instance... When we call `Dispose()` on the process, it naturally disposes of all resources it owns, including those streams.

Additionally, `Dispose()` might abruptly terminate the stream, possibly without having flushed some buffered data. Calling `Close()` instead tends to ensure the buffer is flushed out before the close happens, so the `node` process gets the signal we are trying to send.

File: packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/Runtime.cs (hunk 3 above)
Comment (RomainMuller) [reply packages/@jsii/dotnet-runtime/src/Amazon.JSII.Runtime/Services/Runtime.cs:13]: Previously, `STDERR` from anywhere in `node` (i.e: `console.error` from user's apps) were only surfaced to the console of non-JS applications when `JSII_DEBUG` was set to `true`. This means the behavior of a program was *different* if you ran it "pure JavaScript" versus through a jsii binding. Hence I think this was actually a bug and not a feature.

**Review by eladb — APPROVED** [review]

**Pull request merged**